native/src/routes/SearchModal.tsx

```diff
@@ -37,51 +42,91 @@ const Wrapper = styled.View`
 `
 
 export type SearchModalProps = {
-  categories: CategoriesMapModel
+  allPossibleResults: Array<SearchResult>
   navigateTo: (routeInformation: RouteInformationType) => void
   theme: ThemeType
   languageCode: string
   cityCode: string
   closeModal: (query: string) => void
   t: TFunction<'search'>
   initialSearchText: string
+  loading: boolean
 }
 
 const SearchModal = ({
-  categories,
+  allPossibleResults,
   navigateTo,
   theme,
   languageCode,
   cityCode,
   closeModal,
   t,
   initialSearchText = '',
+  loading,
 }: SearchModalProps): ReactElement => {
   const [query, setQuery] = useState<string>(initialSearchText)
   const resourceCache = useResourceCache({ cityCode, languageCode })
 
-  const searchResults = useMemo(() => searchCategories(categories, query), [categories, query])
-
-  const onItemPress = (category: CategoryModel): void => {
-    const routeInformation: CategoriesRouteInformationType = {
-      route: CATEGORIES_ROUTE,
-      cityContentPath: category.path,
-      cityCode,
-      languageCode,
-    }
-    sendTrackingSignal({
-      signal: {
-        name: SEARCH_FINISHED_SIGNAL_NAME,
-        query,
-        url: urlFromRouteInformation(routeInformation),
+  const minisearch = useMemo(() => {
+    const search = new MiniSearch({
+      fields: ['title', 'content'],
+      storeFields: ['title', 'content', 'path', 'location', 'url', 'thumbnail'],
+      searchOptions: {
+        boost: { title: 2 },
+        fuzzy: true,
+        prefix: true,
       },
     })
-    navigateTo({
-      route: CATEGORIES_ROUTE,
-      cityCode,
-      languageCode,
-      cityContentPath: category.path,
-    })
+    search.addAll(allPossibleResults)
+    return search
+  }, [allPossibleResults])
+
+  // Minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
+  const searchResults = minisearch.search(query) as unknown as SearchResult[]
+
+  const showSnackbar = useSnackbar()
+
+  const followLink = (link: string, isExternalUrl: boolean): void => {
+    if (isExternalUrl) {
+      sendTrackingSignal({
+        signal: {
+          name: SEARCH_FINISHED_SIGNAL_NAME,
+          query,
+          url: link,
+        },
+      })
+      openExternalUrl(link, showSnackbar)
+    } else {
+      const isEvent = link.split('/')[3] === EVENTS_ROUTE
+      const routeInformation: CategoriesRouteInformationType | EventsRouteInformationType = {
+        route: isEvent ? EVENTS_ROUTE : CATEGORIES_ROUTE,
+        cityContentPath: link,
+        cityCode,
+        languageCode,
+      }
```

Comment:
check out https://github.com/digitalfabrik/integreat-app/blob/5dca2f60c5aa23d011145506a451f379ffcc26d5/native/src/hooks/useNavigateToDeepLink.ts#L63-L64 which does exactly what you are doing here

Reply:
Amazing, thanks for the hint!

Reply:
Rather use `useNavigateToLink` which does not include additional deeplinking logic and should be sufficient.